arelle/plugin/validate/DBA/rules/fr.py

```diff
@@ -512,6 +514,173 @@ def rule_fr56(
             )
 
 
+@validation(
+    hook=ValidationHook.XBRL_FINALLY,
+)
+def rule_fr57(
+        pluginData: PluginValidationDataExtension,
+        val: ValidateXbrl,
+        *args: Any,
+        **kwargs: Any,
+) -> Iterable[Validation]:
+    """
+    DBA.FR57: Missing or invalid comparative figures.
+
+    Annual reports, etc. with a start date of 1/1 2016 or later must have a profit
+    allocation, if the profit (fsa:ProfitLoss) for the year is greater than 1000 or
+    less than -1000, cf. the Annual Accounts Act §§ 31 and 95 a.
+    """
+    modelXbrl = val.modelXbrl
+    contexts = pluginData.getReportingPeriodContexts(modelXbrl)
+
+    foundCurrentAssets = None
+    foundCurrentEquity = False
+    foundCurrentLiabilitiesAndEquity = None
+    foundCurrentProfitLoss = False
+    foundPreviousAssets = None
+    foundPreviousEquity = False
+    foundPreviousLiabilitiesAndEquity = None
+    foundPreviousProfitLoss = False
+    assetErrors = []
+    equityErrors = []
+    equalityErrorPairs = []
+    profitLossErrors = []
+    liabilitiesAndEquityErrors = []
+    negativeAssetFacts = []
+    negativeLiabilitiesAndEquityFacts = []
+    if contexts is not None:
+        if "current" not in contexts.keys() and "previous" in contexts.keys():
+            yield Validation.error(
+                codes="DBA.FR57.MarkingOfPeriod",
+                msg=_("The company has marked the previous accounting period, even though it is the first accounting period."),
+            )
+        elif "current" in contexts.keys():
+            assetsFacts = getFactsWithoutDimension(val, pluginData.assetsQn)
+            equityFacts = getFactsWithoutDimension(val, pluginData.equityQn)
+            profitLossFacts = getFactsWithoutDimension(val, pluginData.profitLossQn)
+            liabilitiesAndEquityFacts = getFactsWithoutDimension(val, pluginData.liabilitiesAndEquityQn)
+            currentDurationContextId = contexts["current"].id
+            for profitLoss in profitLossFacts:
+                if profitLoss.context.id == currentDurationContextId:
```

Comment:
Checking against the context ID doesn't seem robust enough. What happens if a filing defines two otherwise identical contexts with different IDs?